semgrep_output_v1.atd

```diff
@@ -1680,6 +1686,42 @@ type ci_scan_complete_stats = {
     <python repr="dict">
     <ts repr="map">
     option;
+
+  (* since 1.98.0 *)
+  (* In collaboration with the Data Science team, it was suggested
+   * that we start to group stats by product for organizational purposes.
+   *)
+  ?supply_chain_stats: supply_chain_stats option;
+}
+
+(* Valid schema versions. Update when making breaking changes. *)
+type stats_schema_version <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True)"> = [
+  | V1_0_0 <json name="1.0.0"> (* initial version *)
```

Comment:
We need a schema version for data warehousing purposes down stream. 

A couple worries:
- We forget to update the schema version on changes
- We'll only ever add a new schema version on breaking changes, so maybe semantic versioning doesn't make sense? Perhaps just a date (`20241112`) would suffice?

Reply:
Agree that there seems to not be much reason to use semantic versioning here. I like the idea of using a date.

In general, though, we don't allow backwards-incompatible changes to the interface types. What defines a breaking change here? Would something like adding a field require updating the version?

Reply:
> Agree that there seems to not be much reason to use semantic versioning here. I like the idea of using a date.

I'll switch to a date then!

>  Would something like adding a field require updating the version? 

No, adding a field would not require updating the version. Basically anytime we remove/change an existing field (so basically anytime the backward compatibility PR check fails). I imagine we won't have too many backward incompatible changes due to the enforcement of backward compatibility for the last X versions. 

Now that I think about it, there’s a case for not restricting version updates solely to breaking changes. It’s really up to us, since the schema version primarily serves to inform the data warehouse (Snowflake) on how to process the incoming stats JSON. 

For instance, if we add a new field and deprecate an old one -- like if `subproject_stats` isn’t sufficient a year from now and we introduce `subproject_stats_v2` -- updating the schema version could guide downstream processing to start using `subproject_stats_v2`. We’d then update the schema version again once we remove `subproject_stats`. 

So, updating on additions or deprecations might make sense to keep data handling in sync.

Reply:
@bkettle changed to:
```
type stats_schema_version <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True)"> = [
  | SSV20241112 (* initial version *)
] 
```

Prefixed with `SSV` since it can't start with a number.

Reply:
@aryx left some good feedback related to schema version. might end up keying off of semgrep version instead. see response below https://github.com/semgrep/semgrep-interfaces/pull/313#discussion_r1844477512

Reply:
Yeah, I think that seems like a good idea